mesonbuild/compilers/rust.py

```diff
@@ -161,6 +161,9 @@ def _native_static_libs(self, work_dir: str, source_name: str) -> None:
                 # no match and kernel == none (i.e. baremetal) is a valid use case.
                 # return and let native_static_libs list empty
                 return
+            if self.info.system == 'emscripten':
+                # no match and emscripten is valid after rustc 1.84
+                return
```

Comment:
If it's expected to always not match, should we skip running `rustc --crate-type staticlib --print .......` entirely, and simply return early from the function depending on `self.version` comparison?